[Vanessa] iP

vanessaxuuan opened the pull request:
# GARY

> "Get busy living or get busy dying." - Stephen King [source](https://atlasmedstaff.com/adventures-in-nursing/get-busy-living-or-get-busy-dying/)

Gary frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here.](https://github.com/vanessaxuuan/ip/tree/A-Jar)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

**And it is FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the **main** method:
```
public class Main {
    public static void main(String[] args) {
        new Gary("./filepath).run();
    }
}
```

Commit:
Add Gradle support

Commit:
level1-gary

Commit:
level2-gary

Commit:
level3-gary

Commit:
Level4-gary

Commit:
Level5-gary

Commit:
Level6-Gary

vanessaxuuan committed:
add memory feature in branch-Level-7

jaysmyname commented on src/main/java/Gary.java:
```diff
@@ -0,0 +1,98 @@
+import java.sql.Array;
+import java.util.*;
```
Shouldn't imported classes be listed explicitly?

jaysmyname commented on src/main/java/Gary.java:
```diff
@@ -0,0 +1,98 @@
+        System.out.println("Bye, hope you stay productive!\n");
+    }
+
+    public static void printList(int type, ArrayList<Task> item) {
```
Perhaps the variable name should be in plural form for ArrayList<Task>?

jaysmyname commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+import java.util.*;
```
Shouldn't imported classes be listed explicitly?

jaysmyname commented on src/main/java/Gary.java:
```diff
@@ -0,0 +1,98 @@
+
+            if (nxt.equals("list")) {
+                Gary.printList(1, items);
+            } else if (nxt.equals("mark")) {
+                System.out.println("which tasks have you completed? e.g. 2 3");
+                String nums = sc.nextLine();
+                String[] first = nums.split(" ");
+                int i = 0;
+                for (String str : first) {
+                    items.get(Integer.parseInt(first[i]) - 1).toMark();
+                    i++;
+                }
+                Gary.printList(2, items);
```
I like that you enabled multi-step commands!

jaysmyname reviewed (commented):
Overall good coding standard practices!
Just remember to import classes explicitly instead of using *

jetrz commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadlines extends Task {
+    String date;
```
It would be good to have an access modifier for these fields. Same comment for Event.

jetrz commented on src/main/java/Gary.java:
```diff
@@ -0,0 +1,98 @@
+            }
+            nxt = sc.nextLine(); // continue getting inputs
+        }
+        System.out.println("Bye, hope you stay productive!\n");
```
Parser is clean and easy to understand!

jetrz commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+    public String toString() {
+        String done = this.isDone ? "X" : " ";
+        return "[" + done + "] " + this.task;
+    }
```
toString is abstracted well for Task and its children classes!

jetrz commented on src/main/java/GaryException.java:
```diff
@@ -0,0 +1,14 @@
+public class GaryException extends Exception {
+    // represent exceptions specific to Gary
+    // int err;
+    String msg;
+
+    public GaryException(String msg) {
+       // this.err = err;
+       this.msg = msg;
+    }
+
+    public void GaryError() {
+        System.out.printf("Sorry, what is %s ?\n", this.msg);
+    }
+}
```
Would be useful to add javadocs documentation for custom exceptions!

jetrz reviewed (commented):
Generally well written code! Minor nitpicks on documentation, but otherwise code implements OOP concepts effectively, and is clean and readable!

vanessaxuuan committed:
add dates and times

vanessaxuuan committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  add dates and times

# Conflicts:
#	src/main/java/Gary.java

vanessaxuuan committed:
use more OOP by extracting classes

vanessaxuuan committed:
divide classes into packages

vanessaxuuan committed:
add JUnit tests

vanessaxuuan committed:
add JavaDoc comments

vanessaxuuan committed:
Add JavaDoc comments to test classes

vanessaxuuan committed:
Add JavaDoc comments

vanessaxuuan committed:
Follow the Coding Standard

vanessaxuuan committed:
Add a method to find a task by searching for a keyword

vanessaxuuan committed:
rebase

vanessaxuuan committed:
Update IP description 

Use GFMD in the PR description

vanessaxuuan committed:
Merge branch 'master' of https://github.com/vanessaxuuan/ip

* 'master' of https://github.com/vanessaxuuan/ip:
  Update IP description

vanessaxuuan committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

* commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d':
  Add Gradle support

vanessaxuuan committed:
Set up gradle

vanessaxuuan committed:
Add GUI to Gary

vanessaxuuan committed:
Add GUI to Gary

vanessaxuuan committed:
Add FXML files

vanessaxuuan committed:
Integrate FXML into GUI for Gary

vanessaxuuan committed:
Add assert features to document important assumptions

Special cases such as,
* empty user input
* operating on an empty to-do list

should be handled specially to prevent errors

Let's add assertions to functions that handle Gary's response

vanessaxuuan committed:
Rename packages

Group classes into packages base on their functionalities for easier management.

Let's move related classes into packages such as,
* gary.task
* gary.ui
* gary.gui
* gary.exception

vanessaxuuan committed:
Improve code quality

Standardise code naming, layout, statements and comments

Add relative file path that works on both Unix and Windows

vanessaxuuan committed:
Merge pull request #3 from vanessaxuuan/branch-A-CodeQuality

Branch a code quality

vanessaxuuan committed:
Merge branch 'master' of https://github.com/vanessaxuuan/ip

* 'master' of https://github.com/vanessaxuuan/ip:
  Improve code quality
  Rename packages

vanessaxuuan committed:
Resolve merge conflicts with master branch

vanessaxuuan committed:
Merge pull request #2 from vanessaxuuan/branch-A-Assertions

Add assert features to document important assumptions

vanessaxuuan committed:
Merge branch 'master' of https://github.com/vanessaxuuan/ip

* 'master' of https://github.com/vanessaxuuan/ip:
  Add assert features to document important assumptions

vanessaxuuan committed:
Update JavaDocs

vanessaxuuan committed:
Finalize GUI

Let's,
* Change display images
* Add welcome message

vanessaxuuan committed:
Merge branch 'branch-better-GUI'

* branch-better-GUI:
  Finalize GUI

vanessaxuuan committed:
Add a representative screenshot of Gary

vanessaxuuan committed:
Add User Guide

fit-for-purpose

vanessaxuuan committed:
Update README.md

vanessaxuuan committed:
Update User Guide

vanessaxuuan committed:
Update README.md

vanessaxuuan committed:
Update README.md

vanessaxuuan committed:
Update README.md

vanessaxuuan committed:
Update README.md

vanessaxuuan committed:
Update README.md

vanessaxuuan committed:
Merge branch 'master' of https://github.com/vanessaxuuan/ip

* 'master' of https://github.com/vanessaxuuan/ip:
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update User Guide
  Update README.md
  Add User Guide
  Add a representative screenshot of Gary

vanessaxuuan committed:
Create Jar file

vanessaxuuan committed:
update jar file

vanessaxuuan committed:
update README.md
